api.bs

```diff
@@ -277,78 +277,138 @@ The aggregation service:
+{{PrivateAttributionConversionOptions}} dictionaries when calling the
+<a method for=PrivateAttribution>saveImpression()</a> or
+<a method for=PrivateAttribution>measureConversion()</a> APIs, respectively.
 
+## Finding a Supported Aggregation Service ## {#list-aggregation-services-api}
 
-## ListAggregationSystems API ## {#list-aggregation-systems-api}
+<p class=issue>Is any additional information required in the
+{{PrivateAttributionAggregationService}} dictionary?
 
-navigator.privateAttribution.listAggregationSystems()
-
-TODO: add whatever else is needed in this struct
+The <dfn method for=PrivateAttribution>listAggregationServices()</dfn> method
+returns a list of aggregation services supported by the [=user agent=]. The page
+must select and specify one of these services when calling the
+<a method for=PrivateAttribution>saveImpression()</a> and
+<a method for=PrivateAttribution>measureConversion()</a> APIs.
 
 <xmp class=idl>
-dictionary PrivateAttributionAggregationSystem {
-  // Name of the aggregation system. This is passed as the `aggregator` parameter to
-  // other APIs.
+dictionary PrivateAttributionAggregationService {
   required DOMString name;
+  required DOMString apiVersion;
+};
+
+[SecureContext, Exposed=Window]
+interface PrivateAttribution {
+  sequence<PrivateAttributionAggregationService> listAggregationServices();
 };
 </xmp>
 
+The arguments to <a method for=PrivateAttribution>listAggregationServices()</a> are as follows:
+
+<dl dfn-for=PrivateAttributionAggregationService dfn-type=dict-member>
+  <dt><dfn>name</dfn></dt>
+  <dd>
+    Name of the aggregation service. This is passed as the `aggregator`
+    parameter to other APIs.
+  </dd>
+  <dt><dfn>apiVersion</dfn></dt>
+  <dd>
+    Version of the Private Attribution API supported by this aggregator. Even if
+    an aggregator supports multiple versions of the API, it is expected to
+    assign a unique aggregation service name for each supported version.
+    Thus, the API version is implicit in the aggregator selection
+    and does not need to be passed to other APIs.
+  </dd>
+</dl>
+
 ## Saving Impressions ## {#save-impression-api}
 
-The <dfn method for=PrivateAttribution>saveImpression()</dfn> method does something or other.
+The <a method for=PrivateAttribution>saveImpression()</a> method requests
+that the [=user agent=] record an [=impression=] in the [=impression store=].
 
 <pre>
 navigator.privateAttribution.saveImpression({
-  aggregator: "aggregator.example", // the name of the aggregation system
-  index: 3,                         // the histogram index for counting this impression
+  aggregator: "aggregator.example",
+  histogramIndex: 3,
   ad: "sample-campaign-eijb",       // a unique identifier for the ad placement
-  target: "advertiser.example",     // the advertiser site where a conversion will occur
+  conversionSite: "advertiser.example",     // the advertiser site where a conversion will occur
 });
 </pre>
```

Comment:
I was considering getting rid of the examples entirely, given that the detailed descriptions of the parameters will include the information that is in the comments here. Do you think it's worth keeping them?

Reply:
I'm a fan of using examples to illustrate how things work.  Maybe I can suggest removing them from here and then adding an examples section at the top of the API section, so that you can give a bit of an overview of how things fit together.

As for the rules... we're not doing an explainer so we should have that sort of explainy stuff in there somewhere.